core/cppgc.rs

```diff
@@ -36,11 +36,37 @@ pub fn make_cppgc_object<'a, T: GarbageCollected + 'static>(
   t: T,
 ) -> v8::Local<'a, v8::Object> {
   let state = JsRuntime::state_from(scope);
-  let templ =
-    v8::Local::new(scope, state.cppgc_template.borrow().as_ref().unwrap());
-  let func = templ.get_function(scope).unwrap();
-  let obj = func.new_instance(scope, &[]).unwrap();
+  let opstate = state.op_state.borrow();
+
+  // To initialize object wraps correctly, we store the function
+  // template in OpState with `T`'s TypeId as the key when binding
+  // because it'll be pretty annoying to propogate `T` generic everywhere.
+  //
+  // Here we try to retrive a function template for `T`, falling back to
+  // the default cppgc template.
+  let id = TypeId::of::<T>();
+  let obj = if let Some(templ) =
```

Comment:
i feel like something is not right. we create a function template which wraps a rust constructor. this function template is exposed to js via the `deno_core_ops_obj`. When we call one of the js constructors from that, it implicitly creates an instance and then calls the rust constructor with `args.this() == that instance`. likewise from rust, we would call `Function::new_instance`, not `ObjectTemplate::new_instance`. so when would this code path be taken?